app/lib/three_scale/analytics/user_tracking.rb

```diff
@@ -26,16 +27,25 @@ def respond_to_missing?(method_sym, include_all)
           end
 
           def method_missing(method_sym, *args, &block)
+            Rails.logger.info("#{method_sym} call avoided. Segment disabled")
```

Comment:
This was made to follow @guicassolato suggestion (https://github.com/3scale/porta/pull/903#discussion_r295384203)

Reply:
Probably only when one of the delegated methods is invoked, no?

Reply:
@guicassolato Then why not to test that the method is invoked in the proper class and that's it?
because that is actually what we want. Like:
`NullAdapter.any_instance.expects(action_name).once`

Reply:
There are 3 different things:
1. `UserTracking::TrackingAdapter.any_instance.expects(action_name).never` – weak but it doesn't really hurt having it. We can keep it in the test.
2. `UserTracking::TrackingAdapter::NullAdapter.any_instance.expects(action_name)` – you are right, @Martouta! This is better than testing the log.
3. Logging message in case the method is invoked when Segment integration is disabled – about this... now I'm thinking... we don't really prevent the method from being called, right? It's just a no-op when called on a `NullAdapter` instance. If this is correct, then better without the logs; otherwise, we will have a bunch of useless information dumped to stdout. On the other hand, if we DO prevent the integration from being used in the first place and those method should not even be invoked, then the logs can help to realize that something is wrong.

Reply:
Like this I mean @guicassolato @duduribeiro : 

![image](https://user-images.githubusercontent.com/11318903/59855671-b0e89500-9375-11e9-8e9f-5b484ce4673b.png)

![image](https://user-images.githubusercontent.com/11318903/59855687-b940d000-9375-11e9-94e5-592042cbf009.png)

Reply:
Makes total sense. Awesome 💯

Reply:
I added the test for NullAdapter expecting something. And I totally agree the expects().never is a weak test because we can easily forget about it.